### PR TITLE
Skip cachix jobs if token is unset

### PR DESCRIPTION
**Pull request by berberman**
We should skip running cachix jobs on PRs opend from forked repos

### CODE REVIEW

**Review by Ailrun — APPROVED** [review]

File: .github/workflows/nix.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
         authToken: ${{ secrets.HLS_CACHIX_AUTH_TOKEN }}
     - if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
       run: nix build
-    - if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
+    - if: ${{ needs.pre_job.outputs.should_skip != 'true' && github.repository_owner == 'haskell' }}
       run: nix develop --profile dev && cachix push haskell-language-server dev
-    - if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
+    - if: ${{ needs.pre_job.outputs.should_skip != 'true' && github.repository_owner == 'haskell' }}
```
Comment (berberman) [inline .github/workflows/nix.yml:51]: mmm, this doesn't seem to work at all

Comment (Ailrun) [reply .github/workflows/nix.yml:51]: How about explicitly check whether the secret is defined or not?

Comment (berberman) [reply .github/workflows/nix.yml:51]: actions/runner/issues/520

Comment (Ailrun) [reply .github/workflows/nix.yml:51]: I guess https://github.com/actions/runner/issues/520#issuecomment-700579336 suggests a way?

Comment (berberman) [reply .github/workflows/nix.yml:51]: I'll try it later. We have to resolve this first, since the nix CI keeps failing, blocking other PRs.

**Pull request merged**